### PR TITLE
using cgroup ns id instead of mnt ns id for path to captured written file

### PR DESCRIPTION
**Pull request by mtcherni95**
Solves #1283

### CODE REVIEW

File: tracee-ebpf/tracee/events_processor.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ func (t *Tracee) processEvent(ctx *context, args map[string]interface{}, argMeta
 			}
 
 			// stop processing if write was already indexed
-			fileName := fmt.Sprintf("%d/write.dev-%d.inode-%d", ctx.MntID, dev, inode)
+			fileName := fmt.Sprintf("%d/write.dev-%d.inode-%d", ctx.CgroupID, dev, inode)
```
Comment (yanivagman) [inline tracee-ebpf/tracee/events_processor.go:103]: Let's use container id here as well

File: tracee-ebpf/tracee/tracee.bpf.c [hunk 2]
```diff
@@ -3471,17 +3471,17 @@ static __always_inline u32 send_bin_helper(void* ctx, struct bpf_map_def *prog_a
     }
 
 #define F_SEND_TYPE   0
-#define F_MNT_NS      (F_SEND_TYPE + sizeof(u8))
-#define F_META_OFF    (F_MNT_NS + sizeof(u32))
+#define F_CGROUP_NS   (F_SEND_TYPE + sizeof(u8))
```
Comment (yanivagman) [inline tracee-ebpf/tracee/tracee.bpf.c:3474]: We are not interested in cgroup namespace but cgroup id

File: tracee-ebpf/tracee/tracee.bpf.c [hunk 3]
```diff
@@ -3471,17 +3471,17 @@ static __always_inline u32 send_bin_helper(void* ctx, struct bpf_map_def *prog_a
...
 #define F_SZ_OFF      (F_META_OFF + SEND_META_SIZE)
 #define F_POS_OFF     (F_SZ_OFF + sizeof(unsigned int))
 #define F_CHUNK_OFF   (F_POS_OFF + sizeof(off_t))
 #define F_CHUNK_SIZE  (MAX_PERCPU_BUFSIZE >> 1)
 
     bpf_probe_read((void **)&(file_buf_p->buf[F_SEND_TYPE]), sizeof(u8), &bin_args->type);
 
-    u32 mnt_id = get_task_mnt_ns_id((struct task_struct *)bpf_get_current_task());
-    bpf_probe_read((void **)&(file_buf_p->buf[F_MNT_NS]), sizeof(u32), &mnt_id);
+    u32 cgroup_id = get_task_cgroup_ns_id((struct task_struct *)bpf_get_current_task());
```
Comment (yanivagman) [inline tracee-ebpf/tracee/tracee.bpf.c:3483]: Wrong helper to get cgroup id.
cgroup namespace id != cgroup id

**Review by yanivagman — COMMENTED** [review]
Other places that should be changed:
https://github.com/aquasecurity/tracee/blob/main/tracee-ebpf/tracee/events_processor.go#L103
https://github.com/aquasecurity/tracee/blob/main/tracee-ebpf/tracee/events_processor.go#L145
https://github.com/aquasecurity/tracee/blob/main/tracee-ebpf/tracee/events_processor.go#L147

File: tracee-ebpf/tracee/events_processor.go (hunk 1 above)
Comment (mtcherni95) [reply tracee-ebpf/tracee/events_processor.go:103]: oops, right

File: tracee-ebpf/tracee/tracee.bpf.c [hunk 4]
```diff
@@ -3471,17 +3471,17 @@ static __always_inline u32 send_bin_helper(void* ctx, struct bpf_map_def *prog_a
...
 #define F_SZ_OFF      (F_META_OFF + SEND_META_SIZE)
 #define F_POS_OFF     (F_SZ_OFF + sizeof(unsigned int))
 #define F_CHUNK_OFF   (F_POS_OFF + sizeof(off_t))
 #define F_CHUNK_SIZE  (MAX_PERCPU_BUFSIZE >> 1)
 
     bpf_probe_read((void **)&(file_buf_p->buf[F_SEND_TYPE]), sizeof(u8), &bin_args->type);
 
-    u32 mnt_id = get_task_mnt_ns_id((struct task_struct *)bpf_get_current_task());
-    bpf_probe_read((void **)&(file_buf_p->buf[F_MNT_NS]), sizeof(u32), &mnt_id);
+    u64 cgroup_id = ((context_t *)ctx)->cgroup_id;
```
Comment (yanivagman) [inline tracee-ebpf/tracee/tracee.bpf.c:3483]: This is NOT the context_t we define in tracee, but the bpf context

**Comment by mtcherni95** [comment]
this PR should be promoted, only after this https://github.com/aquasecurity/tracee/pull/1301

File: tracee-ebpf/tracee/events_processor.go [hunk 5]
```diff
@@ -142,9 +146,13 @@ func (t *Tracee) processEvent(ctx *context, args map[string]interface{}, argMeta
 				}
 				castedSourceFileCtime := int64(sourceFileCtime)
 
-				capturedFileID := fmt.Sprintf("%d:%s", ctx.MntID, sourceFilePath)
+				containerId := t.containers.GetCgroupInfo(ctx.CgroupID).ContainerId
```
Comment (yanivagman) [inline tracee-ebpf/tracee/events_processor.go:149]: You already have containderid initialized above in this function. Why doing it again?

Comment (mtcherni95) [reply tracee-ebpf/tracee/events_processor.go:149]: We do not need it for every case + every time we call the function only one case is matched. I'm okay with taking it apart at the beginning of the switch statement but I don't think is necessary.

**Comment by mtcherni95** [comment]
cherry-picking to https://github.com/aquasecurity/tracee/pull/1301 as concurrency of containers cgroup is need for this current PR only.

**Pull request closed without merging**